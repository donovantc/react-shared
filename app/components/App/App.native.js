/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react'
import { StyleSheet, Text, View, Platform } from 'react-native'
import Example from '../Example'
import Example2 from '../Example2'

class App extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Text>This is the React Native App</Text>
        <Example />
        <Example2 />
      </View>
    )
  }
}

export { App }

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
    justifyContent: 'center',
    alignItems: 'stretch',
    backgroundColor: '#F5FCFF'
  }
})
