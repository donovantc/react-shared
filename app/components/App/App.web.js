import React from 'react'
import Example from '../Example'

const App = props => {
  return (
    <div>
      <h1 style={styles.title}>This is the React Web App</h1>
      <Example />
    </div>
  )
}

export { App }

const styles = {
  title: {
    textAlign: 'center'
  }
}
