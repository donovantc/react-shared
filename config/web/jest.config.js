module.exports = {
  verbose: true,
  rootDir: '../../',
  verbose: true,
  setupFiles: ['<rootDir>/config/testSetup.js'],
  testMatch: ['**/__tests__/*.test.web.js'],
  moduleFileExtensions: ['js', 'jsx', 'json', 'web.js']
}
