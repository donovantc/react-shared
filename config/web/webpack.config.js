const path = require('path');
const HtmlWebpackPlugin = require('html-webpack-plugin');

const HtmlWebpackPluginConfig = new HtmlWebpackPlugin({
    template: './app/index.html',
    filename: 'index.html',
    inject: 'body'
})

module.exports = {
    mode: 'development',
    entry: './app/index.jsx',
    output: {
        path: path.resolve('./dist'),
        filename: 'bundle.js'
    },
    module: {
        rules: [{
            test: /\.(js|jsx|web.js)$/,
            exclude: /node_modules/,
            loader: 'babel-loader'
        }]
    },
    resolve: {
        extensions: ['.js', '.jsx', '.web.js']
    },
    plugins: [HtmlWebpackPluginConfig],
    devServer: {
        port: 3000
    }
};