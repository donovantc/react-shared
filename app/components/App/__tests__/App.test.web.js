import React from 'react'
import { shallow } from 'enzyme';
import { App } from '../App.web'

import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = shallow(<App />)
  expect(tree).toMatchSnapshot()
})
