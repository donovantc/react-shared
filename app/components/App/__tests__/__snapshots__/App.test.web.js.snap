// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          This is the React Web App
        </h1>,
        <ExampleComponent />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This is the React Web App",
          "style": Object {
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "This is the React Web App",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            This is the React Web App
          </h1>,
          <ExampleComponent />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This is the React Web App",
            "style": Object {
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "This is the React Web App",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
